ros2action/ros2action/verb/echo.py

```diff
@@ -0,0 +1,292 @@
+
+        cancel_goal_event_topic = args.action_name + '/_action/cancel_goal/_service_event'
+        cancel_goal_event_msg_type = action_module.Impl.CancelGoalService.Event
+
+        get_result_event_topic = args.action_name + '/_action/get_result/_service_event'
+        get_result_event_msg_type = action_module.Impl.GetResultService.Event
+
+        feedback_topic = args.action_name + '/_action/feedback'
+        feedback_topic_type = action_module.Impl.FeedbackMessage
+
+        status_topic = args.action_name + '/_action/status'
+        status_topic_type = action_module.Impl.GoalStatusMessage
+
+        # Queue for messages from above topic
+        self.output_msg_queue = queue.Queue(args.queue_size)
+
+        run_thread = True
+        # Create a thread to output message from output_queue
+
+        def message_handler():
+            while run_thread:
+                try:
+                    message = self.output_msg_queue.get(block=True, timeout=0.5)
+                    self.output_msg_queue.task_done()
+                except queue.Empty:
+                    continue
+                print(message)
+        output_msg_thread = threading.Thread(target=message_handler)
+        output_msg_thread.start()
+
+        with NodeStrategy(args) as node:
+            send_goal_event_sub = None
+            if ActionInterfaces.GOAL_SERVICE.value in output_interface_list:
+                send_goal_event_sub: Subscription[EventMessage] = node.create_subscription(
+                        send_goal_event_msg_type,
+                        send_goal_event_topic,
+                        self._send_goal_subscriber_callback,
+                        qos_profile_services_default)
+
+            cancel_goal_event_sub = None
+            if ActionInterfaces.CANCEL_SERVICE.value in output_interface_list:
+                cancel_goal_event_sub: Subscription[EventMessage] = node.create_subscription(
+                    cancel_goal_event_msg_type,
+                    cancel_goal_event_topic,
+                    self._cancel_goal_subscriber_callback,
+                    qos_profile_services_default)
+
+            get_result_event_sub = None
+            if ActionInterfaces.RESULT_SERVICE.value in output_interface_list:
+                get_result_event_sub: Subscription[EventMessage] = node.create_subscription(
+                    get_result_event_msg_type,
+                    get_result_event_topic,
+                    self._get_result_subscriber_callback,
+                    qos_profile_services_default)
+
+            feedback_sub = None
+            if ActionInterfaces.FEEDBACK_TOPIC.value in output_interface_list:
+                feedback_sub: Subscription[MsgT] = node.create_subscription(
+                    feedback_topic_type,
+                    feedback_topic,
+                    self._feedback_subscriber_callback,
+                    QoSProfile(depth=10))   # QoS setting refers to action client implementation
```

Comment:
so QoS for these 2 topics also need to be adjusted by action server QoS configuration? these are just topics from action, not service events, that means we need to set QoS separately from service events above? related to above QoS argument extension, probably we need to have more discussion on this. so let's do this as follow-up?

Reply:
Yes. Users can also specify the QoS for feedback and status individually. These QoS settings are different from the QoS for event topics.

Reply:
service events and topic events are being taken and put the message in the queue with SingleThreadedExecutor, and we also have dedicated thread to pop the message data from queue. so it should not require huge queue length? my thought was that we have definition on top with some description to change if necessary, what do you think?

Reply:
> service events and topic events are being taken and put the message in the queue with SingleThreadedExecutor, and we also have dedicated thread to pop the message data from queue. so it should not require huge queue length? my thought was that we have definition on top with some description to change if necessary, what do you think?

Okay.  I think your comment is related to `--queue_size`